src/Sign.Core/CertificateServices/CertificateManagerService.cs

```diff
@@ -0,0 +1,132 @@
+
+            // Check current user certificate store.
+            using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                store.Open(OpenFlags.OpenExistingOnly | OpenFlags.ReadOnly);
+                var certificates = store.Certificates.Find(X509FindType.FindByThumbprint, _Sha1Thumbprint!, false);
+
+                if (certificates.Count == 0)
+                {
+                    throw new ArgumentException(Resources.CertificateNotFound);
+                }
+
+                return Task.FromResult(certificates[0]);
+            }
+        }
+
+
+        [SupportedOSPlatform("windows")] // CsCParameters is Windows-only but we support cross platform frameworks.
+        public async Task<AsymmetricAlgorithm> GetRsaAsync()
+        {
+            ThrowIfUninitialized();
+
+            // Get RSA from a 3rd party cryptographic service provider
+            if (!string.IsNullOrEmpty(_privateMachineKeyContainer))
+            {
+                var cspOptions = new CspParameters();
+
+                cspOptions.KeyContainerName = _privateMachineKeyContainer;
+                cspOptions.Flags = CspProviderFlags.UseMachineKeyStore;
+
+                RSACryptoServiceProvider.UseMachineKeyStore = true;
+
+                return new RSACryptoServiceProvider(cspOptions);
+            }
+            else if (!string.IsNullOrEmpty(_privateKeyContainer))
+            {
+                var cspOptions = new CspParameters();
+
+                cspOptions.KeyContainerName = _privateKeyContainer;
+                cspOptions.Flags = CspProviderFlags.UseDefaultKeyContainer;
+
+                return new RSACryptoServiceProvider(cspOptions);
+            }
+
+            // Extract RSA from Windows Cert Manager certificate.
+            const string RSA = "1.2.840.113549.1.1.1";
+            const string Ecc = "1.2.840.10045.2.1";
+
+            var certificate = await GetCertificateAsync();
+            var keyAlgorithm = certificate.GetKeyAlgorithm();
+
+            switch (keyAlgorithm)
+            {
+                case RSA:
+                    return certificate.GetRSAPrivateKey() ?? throw new InvalidOperationException(Resources.CertificateRSANotFound);
+                case Ecc:
+                    return certificate.GetECDsaPrivateKey() ?? throw new InvalidOperationException(Resources.CertificateECDsaNotFound);
+                default:
+                    throw new InvalidOperationException(Resources.CertificateUnknownSignAlgoError);
+            }
+        }
+
+        public void Initialize(string sha1Thumbprint, string? cryptoServiceProvider, string? privateKeyContainer,
+            string? privateMachineKeyContainer)
+        {
+            this._Sha1Thumbprint = sha1Thumbprint;
+            this._cryptoServiceProvider = cryptoServiceProvider;
+            this._privateKeyContainer = privateKeyContainer;
+            this._privateMachineKeyContainer = privateMachineKeyContainer;
```

Comment:
Added argument validation. Didn't initially since we validate these during argument parsing but it's best to also validate within the class that consumes them.